Add explanation for local verification. (#2309)
optuna/optuna: CONTRIBUTING.md

pytest tests/${TARGET_TEST_FILE_NAME}
```

## Continuous Integration and Local Verification

Optuna repository uses GitHub Actions and CircleCI.

Currently, we are migrating to GitHub Actions but still we use CirclCI for a test of `document`
because it makes it much easier to check built documentation.

### Local Verification

By installing [`act`](https://github.com/nektos/act#installation) and Docker, you can run
tests written for GitHub Actions locally.

```bash
JOB_NAME=checks

> act -j $JOB_NAME

Reviewer: if it only supports `checks`, `act -j checks` is supposed to be simpler.

Reviewer 2: As far as I know, `documentation` and `doctest` work as well.
`documentation` builds documentation with Sphinx on Python 3.8 and `doctest` runs doctest.

Also, without `-j checks`, i.e., only `act` runs `checks`, `documentation`, and `doctest`.

Author: In my env, I can not run `documentation` and `doctest` because virtual env does not have `make` command.

https://github.com/nektos/act#runners

In the above runners, I chosed `Medium Docker Image`. Maybe, that would be the reason.
Which runner did you use for act?

Reviewer 2: I didn't know that. I've used large. So, it seems like the available jobs are up to users' choice of docker for act.

```

Currently, you can run the following jobs:

- `checks`    

>   - Checking the format, coding style, and type hings

Reviewer: `hints`?

>   - Checking the format, coding style, and type hings

Reviewer 2: ```suggestion
  - Checking the format, coding style, and type hints
```

> - `checks`    
>   - Checking the format, coding style, and type hings

Reviewer 2: ```suggestion
- `checks`    
  - Checking the format, coding style, and type hints
- `docuemtnation`
  - Builds documentation including tutorial
- `doctest`
  - Runs doctest
```

>   - Runs unit tests with RDB storages with Python 3.7   

Reviewer 2: This description seems inappropriate as a description of `checks`.

>   - Runs unit tests with RDB storages with Python 3.7   

Reviewer: ```suggestion
  - Runs unit tests with RDB storages with Python 3.7
```

pytest tests/${TARGET_TEST_FILE_NAME}
```

## Continuous Integration and Local Verification

Optuna repository uses GitHub Actions and CircleCI.

Currently, we are migrating to GitHub Actions but still we use CirclCI for a test of `document`
because it makes it much easier to check built documentation.

### Local Verification

By installing [`act`](https://github.com/nektos/act#installation) and Docker, you can run
tests written for GitHub Actions locally.

```bash

> $JOB_NAME=checks

Reviewer: Is it necessary `$`?

act -j $JOB_NAME
```

> Currently, youcan run the following jobs:

Reviewer: ```suggestion
Currently, you can run the following jobs:
```

pytest tests/${TARGET_TEST_FILE_NAME}
```

## Continuous Integration and Local Verification

Optuna repository uses GitHub Actions and CircleCI.

Currently, we are migrating to GitHub Actions but still we use CirclCI for a test of `document`
because it makes it much easier to check built documentation.

### Local Verification

By installing [`act`](https://github.com/nektos/act#installation) and Docker, you can run
tests written for GitHub Actions locally.

```bash
JOB_NAME=checks
act -j $JOB_NAME
```

> Currently, you can run the following jobs:

Reviewer 2: > 
> 
> In my env, I can not run `documentation` and `doctest` because the container does not have `make` command.
> 
> ```
> $ act -j doctest
> ...
> ...
> [Sphinx/doctest]   ✅  Success - Install Dependencies
> [Sphinx/doctest] ⭐  Run Run Doctest
> | /github/workflow/4: line 2: make: command not found
> [Sphinx/doctest]   ❌  Failure - Run Doctest
> Error: exit with `FAILURE`: 127
> ```
> 
> https://github.com/nektos/act#runners
> 
> In the above runners, I chosed `Medium Docker Image`. Maybe, that would be the reason.
> Which runner did you use for act?

As per this, let's make this a little bit more informative.

```suggestion
Currently, you can run the following jobs: `documentation` and `doctest` may not be executable depending on your choice of docker image of act.
```

Author: Thank you for the suggestion! It would be more kind to users.
